FRCore/FRCore/Keychain/SecuredKey.swift

```diff
@@ -180,17 +180,25 @@ public struct SecuredKey {
     
     /// Decrypts Data object using SecuredKey object
     /// - Parameter data: Decrypted Data object
-    public func decrypt(data: Data) -> Data? {
+    public func decrypt(data: Data, secAlgorithm: SecKeyAlgorithm = .eciesEncryptionCofactorVariableIVX963SHA256AESGCM) -> Data? {
         
-        guard SecKeyIsAlgorithmSupported(privateKey, .decrypt, algorithm) else {
-            Log.e("\(algorithm) is not supported on the device.")
+        guard SecKeyIsAlgorithmSupported(privateKey, .decrypt, secAlgorithm) else {
+            Log.e("\(oldAlgorithm) is not supported on the device.")
             return nil
         }
         
         var error: Unmanaged<CFError>?
-        let decryptedData = SecKeyCreateDecryptedData(privateKey, algorithm, data as CFData, &error) as Data?
+        let decryptedData = SecKeyCreateDecryptedData(privateKey, secAlgorithm, data as CFData, &error) as Data?
         if let error = error {
-            Log.e("Failed to decrypt data: \(error)")
+            Log.e("Failed to decrypt data -  attempting Legacy Algorithm: \(error)")
+            var decryptError: Unmanaged<CFError>?
+            let decryptedData = SecKeyCreateDecryptedData(privateKey, oldAlgorithm, data as CFData, &decryptError) as Data?
```

Comment:
why do we need to keep the old algorithm ? - i believe its for backward compatibility , mainly whoever already has the key encrypted using old algorithm still has to decrypt